### PR TITLE
Add esVisibiltyManager for list workflow APIs on ElasticSearch

### PR DESCRIPTION
**Pull request by vancexu**
This PR add an implementation for VisibilityManager on ES, so that users can make use of visibility data on ElasticSearch using existing list APIs.

### CODE REVIEW

File: service/worker/indexer/esProcessor.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func NewESProcessorAndStart(config *Config, client *elastic.Client, processorNam
 		metricsClient: metricsClient,
 	}
 
-	processor, err := client.BulkProcessor().
+	processor, err := client.GetRawClient().BulkProcessor().
```
Comment (samarabbas) [inline service/worker/indexer/esProcessor.go]: Do not expose the entire raw client.  You can just expose BulkProcessor.

File: service/frontend/workflowHandler.go [hunk 2]
```diff
@@ -3112,3 +3119,13 @@ func (wh *WorkflowHandler) bucketName(customBucketName *string) string {
 func (wh *WorkflowHandler) customBucketNameProvided(customBucketName *string) bool {
 	return customBucketName != nil && len(*customBucketName) != 0
 }
+
+func (wh *WorkflowHandler) getVisibilityManager(domain string) persistence.VisibilityManager {
```
Comment (samarabbas) [inline service/frontend/workflowHandler.go]: Can we wrap this under a wrapper implementation which also implements VisibilityMgr interface.

File: common/persistence/elasticsearchVisibility.go [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+}
+
+func (v *esVisibilityManager) Close() {}
+
+func (v *esVisibilityManager) GetName() string {
+	return esPersistenceName
+}
+
+func (v *esVisibilityManager) RecordWorkflowExecutionStarted(request *RecordWorkflowExecutionStartedRequest) error {
+	return errOperationNotSupported
```
Comment (samarabbas) [inline common/persistence/elasticsearchVisibility.go]: Can we have Kafka publisher code moved here.

File: service/worker/indexer/esProcessor.go (hunk 1 above)
Comment (vancexu) [reply service/worker/indexer/esProcessor.go]: updated

File: service/frontend/workflowHandler.go (hunk 2 above)
Comment (vancexu) [reply service/frontend/workflowHandler.go]: good suggestion, updated.

File: common/persistence/elasticsearchVisibility.go (hunk 3 above)
Comment (vancexu) [reply common/persistence/elasticsearchVisibility.go]: ok, will do in separate PR.

**Pull request merged**